File: U_Net/utils/polar_to_complex.py

__all__ = ["polar_to_complex"]

from numpy import cos, sin
from numpy import ndarray

def polar_to_complex(array : ndarray) : 
    """Make polar-coordinated array to complex-valued array.

    If input shape of input array is `(a, b, ..., c, 2)`, the return shape will be `(a, b, ..., c)`.

    The `(a, b, ..., c, 0)` element will be the radius in complex plane, 
    and `(a, b, ..., c, 1)` element will be radian angle.

    Args:
        array (ndarray): input array

    Returns:
        complex_array (ndarray): complex-valued array

    ---
    ## Example
    ```python
    >>> A = np.array([
    ...     [1, 0],
    ...     [5, pi]
    ... ])
    >>> polar_to_complex(A)
    array([ 1.+0.000000e+00j, -5.+6.123234e-16j])
    
    ```

    """
    r = array[..., 0]
    theta = array[..., 1]

    real_part = r * cos(theta)
    imag_part = r * sin(theta)
    
    complex_array = real_part + imag_part * 1j

    return complex_array

File: MATLAB code/essential/polar_to_complex.m

% Function to convert polar coordinates to complex numbers
function result_array = polar_to_complex(array)
    r = array(:, :, 1);                             % Assign distance
    theta = array(:, :, 2);                         % Assign angle
    real_part = r.*cos(theta);                      % Calculate the real part
    imag_part = r.*sin(theta);                      % Calculate the imaginary part
    
    result_array = complex(real_part, imag_part);   % Assign complex number
end